algebird-core/src/main/scala/com/twitter/algebird/BloomFilter.scala

```diff
@@ -150,6 +154,8 @@ case class BFItem(item: String, hashes: BFHash, width: Int) extends BF {
 
   def +(other: String) = this ++ BFItem(other, hashes, width)
```

Comment:
An extra "if <string equality>" shortcut is definitely cheaper than hashes calculations - benefit of this trade-off will depend on data consumed. Probably doesn't hurt to put the check there - but as you said, it would happens pretty rarely - data values must be same. Your call.